### PR TITLE
[10.0][IMP] purchase request cancelled state

### LINKED ISSUE #396: [9.0][purchase_request_procurement] Procurement not cancelled when a purchase request is cancelled
We need to change the purchase_request_procurement module, so that cancelling a purchase request will cancel the procurement as well. But then you should NOT be able to reset the purchase request back to draft, if the related procurements are cancelled.

Also, you should not be able to cancel a procurement if a subsequent purchase request exists. The user needs to first cancel it through the procurement.

### PR DESCRIPTION
**Pull request by benwillig**
Fixes #396.

From the work of #412 and #416.

### CODE REVIEW

**Comment by benwillig** [comment]
@lreficent Your review is welcome.

**Review by LoisRForgeFlow — APPROVED** [review]
LGTM :+1: 

Thanks for the work!

**Comment by benwillig** [comment]
@lreficent I added a commit. I think we should not copy cancelled field when duplicating a line.

**Comment by LoisRForgeFlow** [comment]
@benwillig mmm I was about to tell that it wasn't very important since the only way to duplicate lines is duplicating the PR itself, but you can actually duplicate a line :open_mouth:. I think that shouldn't be possible, don't  you think? every created/new line must pass through the approval mechanism.

**Comment by benwillig** [comment]
The case when you only duplicate the line may not appear often. But as you said, when you duplicate a PR, it copies the lines. So if a user duplicate a rejected PR, PR lines shouldn't be cancelled.

**Comment by LoisRForgeFlow** [comment]
@benwillig this is fixed in v9 here: https://github.com/OCA/purchase-workflow/pull/428

**Review by MiquelRForgeFlow — APPROVED** [review]
LGTM :+1:

**Review by ThomasBinsfeld — APPROVED** [review]

File: purchase_request/views/purchase_request_view.xml [hunk 1]
```diff
@@ -247,7 +251,7 @@
             <field name="model">purchase.request.line</field>
             <field name="priority" eval="20"/>
             <field name="arch" type="xml">
-                <form string="Purchase Request Line" version="7.0">
+                <form string="Purchase Request Line">
```
Comment (LoisRForgeFlow) [inline purchase_request/views/purchase_request_view.xml:254]: I would update to `<form string="Purchase Request Line" duplicate="false">`

**Review by LoisRForgeFlow — COMMENTED** [review]
The fix related to my https://github.com/OCA/purchase-workflow/pull/424#issuecomment-312221065 is partially attended. I point what is missing in this review

**Pull request merged**